test/unit/alignment/aligned_sequence_test_template.hpp

```diff
@@ -29,108 +31,186 @@ TYPED_TEST_P(aligned_sequence, fulfills_concept)
+
+    auto it = insert_gap(aligned_seq, std::next(begin(aligned_seq), 1), 0);
+
+    typename TypeParam::value_type val{'C'_dna4};
+    val = 'C'_dna4;
+    EXPECT_EQ(*it, val);
+    EXPECT_TRUE((ranges::equal(aligned_seq, seq)));
 }
 
 TYPED_TEST_P(aligned_sequence, erase_one_gap)
 {
     // 1) Removing an actual gap
     TypeParam aligned_seq;
-    TypeParam aligned_seq_expected;
-    TestFixture::initialize_typed_test_container(aligned_seq, "A-CTA");
-    TestFixture::initialize_typed_test_container(aligned_seq_expected, "ACTA");
+    TestFixture::initialise_typed_test_container(aligned_seq, seq);
+    EXPECT_EQ(aligned_seq.size(), 4u);
+    insert_gap(aligned_seq, std::next(begin(aligned_seq), 1));
+    EXPECT_EQ(aligned_seq.size(), 5u);
 
-    auto it = erase_gap(aligned_seq, begin(aligned_seq) + 1);
+    auto it = erase_gap(aligned_seq, std::next(begin(aligned_seq), 1));
+    EXPECT_EQ(aligned_seq.size(), 4u);
 
     typename TypeParam::value_type val{'C'_dna4};
     val = 'C'_dna4;
     EXPECT_EQ(*it, val);
-    EXPECT_EQ(aligned_seq, aligned_seq_expected);
 
     // 2) Removing a non-gap
-    TypeParam aligned_seq_fail;
-    TestFixture::initialize_typed_test_container(aligned_seq_fail, "A-CTA");
+    EXPECT_THROW(erase_gap(aligned_seq, std::next(begin(aligned_seq), 2)), gap_erase_failure);
 
-    EXPECT_THROW(erase_gap(aligned_seq_fail, begin(aligned_seq_fail) + 2), gap_erase_failure);
-
-    EXPECT_EQ(aligned_seq_fail[1], gap{}); // nothing has changed
+    EXPECT_EQ(aligned_seq.size(), 4u); // no change
 }
 
 TYPED_TEST_P(aligned_sequence, erase_multiple_gaps)
 {
-    // 1) Removing actual gaps
-
-    // nucleotide alphabet
     TypeParam aligned_seq;
-    TypeParam aligned_seq_expected;
-    TestFixture::initialize_typed_test_container(aligned_seq, "A--CTA");
-    TestFixture::initialize_typed_test_container(aligned_seq_expected, "ACTA");
 
-    auto it = erase_gap(aligned_seq, begin(aligned_seq) + 1, begin(aligned_seq) + 3);
+    // 1) Removing a gap of length > 1
+    TestFixture::initialise_typed_test_container(aligned_seq, seq);
+    EXPECT_EQ(aligned_seq.size(), 4u);
+    insert_gap(aligned_seq, std::next(begin(aligned_seq), 1), 2);
+    EXPECT_EQ(aligned_seq.size(), 6u);
+
+    auto it = erase_gap(aligned_seq, std::next(begin(aligned_seq), 1), std::next(begin(aligned_seq), 3));
 
+    EXPECT_EQ(aligned_seq.size(), 4u);
     typename TypeParam::value_type val{'C'_dna4};
     val = 'C'_dna4;
     EXPECT_EQ(*it, val);
-    EXPECT_EQ(aligned_seq, aligned_seq_expected);
 
     // 2) Removing a non-gap
-    TypeParam aligned_seq_fail;
-    TestFixture::initialize_typed_test_container(aligned_seq_fail, "A-CTA");
+    TestFixture::initialise_typed_test_container(aligned_seq, seq); // reset
 
-    EXPECT_THROW(erase_gap(aligned_seq_fail, begin(aligned_seq_fail) + 1, begin(aligned_seq_fail) + 3),
+    EXPECT_THROW(erase_gap(aligned_seq, std::next(begin(aligned_seq), 1), std::next(begin(aligned_seq), 3)),
```

Comment:
I would recommen `std::ranges::next`, I know it's tedious. Maybe should think about giving up our own aliases and just offering `namespace seqan3::_sr` as an alias for `std::ranges`. That way we could just right `_sr::next` `_sr::begin` ... everywhere.

Reply:
whats wrong with std::next?

Reply:
https://en.cppreference.com/w/cpp/iterator/next:
"ForwardIt must meet the requirements of LegacyForwardIterator. "

 We just established above that the exact iterator you are testing here does not. :roll_eyes:

Reply:
:see_no_evil: